docker-entrypoint/docker-entrypoint.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+	//For testing purposes
+	//"k8s.io/kubernetes/pkg/client/restclient"
+
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	WAITING  = "waiting"
+	READY    = "ready"
+	RUNNING  = "Running"
+	INTERVAL = 2
+)
+
+var (
+	//"Info logger""
+	Info *log.Logger
+	//"Error logger"
+	Error *log.Logger
+)
+
+type dependency interface {
```

Comment:
You can then create a folder called `dependencies` containing every supported deps in their own sub-folders (with their type definition: with namespace/name strings, a constructor, init(), `GetType()` and `IsResolved()`). In `docker-entrypoint/`, you can have a `dependency.go` that will have a `RegisterDependency` function, called by the init() function of each dep.